app.go

```diff
@@ -82,22 +83,47 @@ func (app *KatzenmintApplication) isTxValid(rawTx []byte) (tx *Transaction, err
 	switch tx.Command {
 	case PublishMixDescriptor:
 		var verifier cert.Verifier
-		payload := DecodeHex(tx.Payload)
+		payload = DecodeHex(tx.Payload)
 		verifier, err = s11n.GetVerifierFromDescriptor(payload)
 		if err != nil {
 			err = ErrTxDescInvalidVerifier
 			return
 		}
-		_, err = s11n.VerifyAndParseDescriptor(verifier, payload, tx.Epoch)
+		desc, err = s11n.VerifyAndParseDescriptor(verifier, payload, tx.Epoch)
 		if err != nil {
 			err = ErrTxDescFalseVerification
 			return
 		}
+		if !app.state.isDescriptorAuthorized(desc) {
+			err = ErrTxDescNotAuthorized
+			return
+		}
+
 	case AddConsensusDocument:
+		payload = []byte(tx.Payload)
+		doc, err = s11n.VerifyAndParseDocument(payload)
+		if err != nil {
+			err = ErrTxDocFalseVerification
+			return
+		}
+		if doc.Epoch != tx.Epoch {
```

Comment:
Should we remove Epoch in tx? Thought this might be redundant.

Reply:
Currently we still need tx.epoch when posting a descriptor. If we put such information in the descriptor structure then I think tx.epoch could be removed.